### PR TITLE
Delete Rundown and Generate Rundown with AI

### PR DESCRIPTION
**Pull request by swastikanata**

### CODE REVIEW

File: src/app/event/[eventId]/(eventId)/rundown/RundownTable.tsx [hunk 1]
```diff
@@ -24,12 +33,46 @@ export const RundownTable = ({ eventId }: RundownTableProps) => {
     setOpenPrompt(true)
   }
 
+  const handleDelete = async (rundownId: string) => {
+    try {
+      await deleteRundown({ id: rundownId }).unwrap()
+    } catch (error) {
+      console.error('Failed to delete the rundown', error)
```
Comment (catherineeangel) [inline src/app/event/[eventId]/(eventId)/rundown/RundownTable.tsx:40]: I think it is better to use toast to make sure user get the error message. also if there is an error message from the backend you should show that instead to give better understanding (if any). example:
```
toast.error("message here")
```

File: src/components/elements/Button/style.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export const ButtonVariants = cva(
         secondary: 'bg-teal-400 text-white hover:bg-teal-300',
         ghost:
           'text-teal-400 hover:text-teal-600 border border-teal-400 hover:border-teal-400',
-        danger: 'text-rose-600 text-white hover:bg-rose-400 bg-rose-500',
+        danger: 'bg-rose-600 text-white hover:bg-rose-400 bg-rose-500',
```
Comment (catherineeangel) [inline src/components/elements/Button/style.ts:12]: there is a duplicate bg color, i think you can remove the later.

**Review by catherineeangel — CHANGES_REQUESTED** [review]

File: src/app/event/[eventId]/(eventId)/rundown/RundownTable.tsx (hunk 1 above)
Comment (JohannesSetiawan) [reply src/app/event/[eventId]/(eventId)/rundown/RundownTable.tsx:40]: I agree with catherine. Use toast.error() for displaying any message of potential error.

File: src/app/event/[eventId]/(eventId)/rundown/RundownTable.tsx [hunk 3]
```diff
@@ -24,12 +33,46 @@ export const RundownTable = ({ eventId }: RundownTableProps) => {
...
+    }
+  }
+
+  const handleDeleteAll = async (eventId: string) => {
+    handleCloseDeleteAllPopup()
+    try {
+      await deleteAllRundown({ eventId: eventId }).unwrap()
+    } catch (error) {
+      console.error('Failed to delete all task steps', error)
```
Comment (JohannesSetiawan) [inline src/app/event/[eventId]/(eventId)/rundown/RundownTable.tsx:49]: This part here too, same as above. Use toast.error()

**Review by JohannesSetiawan — CHANGES_REQUESTED** [review]
Good implementation, but some parts can be changed to create a better app.

**Review by JohannesSetiawan — APPROVED** [review]

File: src/redux/api/rundownApi.ts [hunk 4]
```diff
@@ -42,11 +42,40 @@ export const rundownApi = baseApi.injectEndpoints({
       }),
       invalidatesTags: (result) => [{ type: 'Rundown', id: result?.id }],
     }),
+    deleteRundown: builder.mutation<void, { id: string }>({
+      query: ({ id }) => ({
+        url: `/rundowns/${id}`,
+        method: 'DELETE',
+      }),
+      invalidatesTags: ['Rundown'],
```
Comment (emiriko) [inline src/redux/api/rundownApi.ts:50]: For caching you could invalidate, not the whole rundown but just the rundown for that ID. This way the system doesn't have to refetch the whole thing and just fetch for 1 detail rundown. 

What I meant is that something like this:

```
invalidatesTags:(_, __, arg) => [
        { type: 'Rundown', id: arg.id },
        'Timeline',
      ],
```

**Review by emiriko — CHANGES_REQUESTED** [review]
Overall your implementation is really nice! Next time you can add the description so that we know what you are working on!

**Review by emiriko — APPROVED** [review]
LGTM!

**Review by nataniadeandra — APPROVED** [review]
LGTM

**Pull request merged**